test.sh

```diff
@@ -78,7 +78,7 @@ function run_unit_tests() {
     # Run the full suite of tests once with the -race flag. Since this isn't
     # running tests individually we can't collect coverage information.
     echo "running test suite with race detection"
-    go test -race -p 1 ${TESTPATHS}
+    go test -race -p 1 ${TESTPATHS} || FAILURE=1
```

Comment:
Let's follow the same pattern as the `else` clause here and prefix with `run` rather than `|| FAILURE=1`. While you're at it, could you adjust `run_test_coverage` to do the same?